src/module.c

```diff
@@ -958,9 +958,15 @@ int TSDB_get(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     } else {
         series = RedisModule_ModuleTypeGetValue(key);
     }
-    RedisModule_ReplyWithArray(ctx, 2);
-    RedisModule_ReplyWithLongLong(ctx, series->lastTimestamp);
-    RedisModule_ReplyWithDouble(ctx, series->lastValue);
+    
+    if(SeriesGetNumSamples(series)==0){
```

Comment:
This is an expansive operation.
It is O(chunks)

Reply:
Should use instead -
`u_int64_t(*GetNumOfSample)(Chunk_t *chunk)`
Only the first chunk is created preemptively. Later, we create chunks only when a sample do not has space in previous chunks.

Reply:
> Should use instead -
> `u_int64_t(*GetNumOfSample)(Chunk_t *chunk)`
> Only the first chunk is created preemptively. Later, we create chunks only when a sample do not has space in previous chunks.
Agree. but we should alter behaviour to match O(1)
I think we should alter Series to contain the count of elements in it. It will improve `TS.INFO` `TS.GET` and potentially others that rely on the count of elements.

Reply:
@ashtul @gkorland this removes the requirement of iterating on all chuncks for counting total samples. Improves TS.GET and TS.INFO

Reply:
This solution doesn't take care of retired chunks/samples :(
If we want to go down this path you should change `SeriesTrim` to deduct number of samples in retired slices.

Reply:
@ashtul please revise the latest commit. Also extending the test_trim :)